test/e2e/memcached_test.go

```diff
@@ -76,23 +81,127 @@ func TestMemcached(t *testing.T) {
 	}
 
 	t.Log("Generating k8s")
-	cmdOut, err = exec.Command("operator-sdk",
-		"generate",
-		"k8s").CombinedOutput()
+	cmdOut, err = exec.Command("operator-sdk", "generate", "k8s").CombinedOutput()
 	if err != nil {
 		t.Fatalf("Error: %v\nCommand Output: %s\n", err, string(cmdOut))
 	}
 
+	// create crd
+	crdYAML, err := ioutil.ReadFile("deploy/crd.yaml")
+	err = e2eutil.CreateFromYAML(t, crdYAML, f.KubeClient, f.KubeConfig, "")
+	if err != nil {
+		t.Fatal(err)
+	}
+	ctx.AddFinalizerFn(func() error {
+		extensionclient, err := extensions.NewForConfig(f.KubeConfig)
+		if err != nil {
+			return err
+		}
+		err = extensionclient.ApiextensionsV1beta1().CustomResourceDefinitions().Delete("memcacheds.cache.example.com", metav1.NewDeleteOptions(0))
+		if err != nil && !apierrors.IsNotFound(err) {
+			return err
+		}
+		return nil
+	})
+	t.Log("Created crd")
+
+	// run both subtests
+	t.Run("memcached-group", func(t *testing.T) {
+		t.Run("Cluster", MemcachedCluster)
+		t.Run("Local", MemcachedLocal)
+	})
+}
+
+func memcachedScaleTest(namespace string, f *framework.Framework, t *testing.T) error {
+	// create example-memcached yaml file
+	err := ioutil.WriteFile("deploy/cr.yaml",
+		[]byte("apiVersion: \"cache.example.com/v1alpha1\"\nkind: \"Memcached\"\nmetadata:\n  name: \"example-memcached\"\nspec:\n  size: 3"),
+		os.FileMode(filemode))
+	if err != nil {
+		return err
+	}
+
+	// create memcached custom resource
+	crYAML, err := ioutil.ReadFile("deploy/cr.yaml")
+	e2eutil.CreateFromYAML(t, crYAML, f.KubeClient, f.KubeConfig, namespace)
+
+	// wait for example-memcached to reach 3 replicas
+	err = e2eutil.DeploymentReplicaCheck(t, f.KubeClient, namespace, "example-memcached", 3, 6)
+	if err != nil {
+		return err
+	}
+
+	// update memcached CR size to 4
+	memcachedClient := e2eutil.GetCRClient(t, f.KubeConfig, crYAML)
+	err = memcachedClient.Patch(types.JSONPatchType).
+		Namespace(namespace).
+		Resource("memcacheds").
+		Name("example-memcached").
+		Body([]byte("[{\"op\": \"replace\", \"path\": \"/spec/size\", \"value\": 4}]")).
+		Do().
+		Error()
+	if err != nil {
+		return err
+	}
+
+	// wait for example-memcached to reach 4 replicas
+	err = e2eutil.DeploymentReplicaCheck(t, f.KubeClient, namespace, "example-memcached", 4, 6)
+	return err
+}
+
+func MemcachedLocal(t *testing.T) {
+	t.Parallel()
 	// get global framework variables
 	f := framework.Global
+	ctx := f.NewTestCtx(t)
+	defer ctx.Cleanup(t)
+	namespace, err := ctx.CreateNamespace(f, t)
+	if err != nil {
+		t.Fatal(err)
+	}
+	cmd := exec.Command("operator-sdk", "up", "local", "--namespace="+namespace)
+	stderr, _ := os.Create("stderr.txt")
+	cmd.Stderr = stderr
+	defer stderr.Close()
+
+	err = cmd.Start()
+	if err != nil {
+		t.Fatalf("Error: %v", err)
+	}
+	ctx.AddFinalizerFn(func() error { return cmd.Process.Signal(os.Interrupt) })
```

Comment:
Does `os.Interrupt` actuall kill the up local process? I thought that it simply ignores `Interrupt` signal based on https://github.com/operator-framework/operator-sdk/issues/366.

Reply:
It only stops the `operator-sdk` process, not the `go run` process that `operator-sdk` spawns. It will be this way until #366 gets fixed, as both killing and interrupting both leave the `go run` process running and there's not much we can really do about that.